yt_dlp/extractor/youtube.py

```diff
@@ -5505,6 +5505,109 @@ def _real_extract(self, url):
             ie=YoutubeTabIE.ie_key())
 
 
+class YoutubeNotificationsIE(YoutubeBaseInfoExtractor):
+    IE_NAME = 'youtube:notifications'
+    IE_DESC = 'YouTube notification menu videos; ":ytnotif" keyword (requires cookies)'
+    _VALID_URL = r':ytnotif(?:ication)?s?(:(?P<max_pages>\d+))?'
+    _LOGIN_REQUIRED = True
+    _TESTS = [{
+        'url': ':ytnotif',
+        'only_matching': True,
+    }, {
+        'url': ':ytnotifications:2',
+        'only_matching': True,
+    }]
+
+    @staticmethod
+    def _extract_notification_menu(response, continuation_list):
+        notification_list = traverse_obj(
+            response, ('actions', 0, 'openPopupAction', 'popup', 'multiPageMenuRenderer', 'sections', 0, 'multiPageMenuNotificationSectionRenderer', 'items'), expected_type=list)
+        if not notification_list:
+            notification_list = traverse_obj(
+                response, ('actions', 0, 'appendContinuationItemsAction', 'continuationItems'), expected_type=list)
+        for item in notification_list:
+            for key in ('notificationRenderer', 'continuationItemRenderer'):
+                value = item.get(key)
+                if value:
+                    if key == 'notificationRenderer':
+                        yield value
+                        break
+                    elif key == 'continuationItemRenderer':
+                        continuation_list[0] = value
+                        return
+        continuation_list[0] = None
+
+    def _extract_notification(self, notification):
+        # notificationRenderer
+        video_id = traverse_obj(
+            notification, ('navigationEndpoint', 'watchEndpoint', 'videoId'), expected_type=str)
+        if not video_id:
+            return None
+        thumbnails = traverse_obj(notification, ('videoThumbnail', 'thumbnails'), expected_type=list)
+        status_text = traverse_obj(notification, ('shortMessage', 'simpleText'), expected_type=str)
+        uploader = traverse_obj(
+            notification, ('contextualMenu', 'menuRenderer', 'items', 1, 'menuServiceItemRenderer', 'text', 'runs', 1, 'text'), expected_type=str)
+        status_text = status_text.removeprefix(uploader + ' ')
+        split_text = status_text.split(': ', 2)
+        title = None
+        if len(split_text) == 2:
+            title = split_text[1]
+        extra_info = {
+            'thumbnails': thumbnails,
+            'uploader': uploader,
+        }
+        return self.url_result(
+            f'https://www.youtube.com/watch?v={video_id}', ie=YoutubeIE.ie_key(),
+            video_id=video_id, video_title=title, **extra_info)
```

Comment:
```suggestion
        return self.url_result(
            f'https://www.youtube.com/watch?v={video_id}', YoutubeIE,
            video_id, title, thumbnails=thumbnails, uploader=uploader)
```
or, if we end up extracting more fields, it's better to do:
```py
        return {
            '_type': 'url',
            'url': f'https://www.youtube.com/watch?v={video_id}',
            'ie_key': YoutubeIE.ie_key(),
            'video_id': video_id,
            'title': title,
            ...
        }
```